lib/ssz_ex.ex

```diff
@@ -367,26 +377,75 @@ defmodule LambdaEthereumConsensus.SszEx do
     end
   end
 
-  defp decode_bitvector(bit_vector, size) when bit_size(bit_vector) == size,
-    do: {:ok, BitVector.new(bit_vector, size)}
+  defp decode_bitlist(_bit_list, _max_size), do: {:error, "invalid bitlist"}
+
+  defp decode_bitvector(bit_vector, size) do
+    mask = 255 >>> (8 - rem(size, 8) &&& 7)
+    num_bytes = byte_size(bit_vector)
+
+    cond do
+      size == 0 and num_bytes == 1 ->
+        <<bit::integer-size(1), _rest::bitstring>> = bit_vector
+
+        if bit == 0 do
+          {:ok, BitVector.new(bit_vector, size)}
+        else
+          {:error, "ExcessBits"}
+        end
+
+      size == 0 and num_bytes == 0 ->
+        {:error, "ExcessBits"}
```

Comment:
According to the specs:

> Empty vector types (`Vector[type, 0]`, `Bitvector[0]`) are illegal.

Can't we just return an error on `size == 0` regardless of the content?